Code_In_Progress/MyTextWebscraperV3.py

```diff
@@ -0,0 +1,188 @@
+            first_column = row.findAll('td')[0].contents
+            archive_column_numbers.append(first_column)    # finds the numbers for the archives url, throws into list
+        archive_column_numbers.pop(0)
+        count_of_archive_numbers=len(archive_column_numbers)
+        print (count_of_archive_numbers)
+
+        #count_down=count_of_archive_numbers
+        count_down=2999
+        string1=''
+        #for i in range(count_of_archive_numbers):     # takes last number in list, goes to each irl in list
+
+        for i in range(2999):      #the amount of messages you want. up to 3000
+            try:
+                if i == 500:
+                    time.sleep(300)
+                if i == 1000:
+                    time.sleep(300)
+                if i == 1500:
+                    time.sleep(300)
+                if i == 2000:
+                    time.sleep(300)
+                if i == 2500:
+                    time.sleep(300)
+                output=''
+                string1=''
+                url=''
+                archive_attach=archive_column_numbers[count_down]
+                count_down-=1
+                for ele in archive_attach:     #makes list into string for url
+                    string1+=archive_attach[0]
+                print("pass", i)
+                url = 'https://boards.4channel.org/pol/thread/'+str(string1)
+                print(url)
+                res = requests.get(url)
+                print(res)
+                html_page = res.content
+                soup = BeautifulSoup(html_page, 'html.parser')
+                text = soup.find_all(text=True)
+                print(text)
+            except:
+                text=""
+                input("An error occured while scraping. Type anything to continue")    #stops it from running if error occurs(for internet issues)
+
+
+           # blacklist = [
+            #    '[document]',
+            #    'noscript',
+            #    'header',
+            #    'html',
+            #    'meta',
+            #    'head',
+            #    'input',
+            #    'script',
+            #    # there may be more elements you don't want, such as "style", etc.
+            #]
+
+            for t in text:
+                #print(t.parent.name)
+
+                if t.parent.name == "blockquote":
+                    output += '{} '.format(t)
+
+            print(output)
+            output2=output
+            output2 = output2.encode("utf-8")
+            print("v")
+
+
+            file1.write(str(output2)+'\n')
+
+
+
+            newstring+=output
+            #print("this is new string", newstring)
+            time.sleep(5)
+
+    #output=output.split()            #seperates messages into words
+
+    newstring = re.findall(r"[\w']+", newstring)    #seperates messages into words and deals woth punctuation
+
+# printing result
+    #print("The list after performing split functionality : " , newstring)
+
+
+    for word in newstring:
```

Comment:
All your text analysis code should happen in a function separate from the webscraper code. The reason is so that you can 1) modularize your code and reduce dependencies, 2) you can run analysis on text 'offline' rather than only at the time when you drag if off the webpage, 3) if you want to visualize this data, you'll need to save it somewhere anyway

Reply:
Your webscraper code will also run faster